libsodium-sys/build.rs

```diff
@@ -63,34 +94,417 @@ fn find_libsodium_env() -> String {
     } else {
         println!("cargo:rustc-link-lib={0}=sodium", mode);
     }
+}
+
+/* Must be called when SODIUM_LIB_DIR or SODIUM_INC_DIR is set to any value
+This function will set `cargo` flags.
+Return: SODIUM_INC_DIR
+*/
+fn find_libsodium_env() -> Option<String> {
+    let lib_dir = env::var("SODIUM_LIB_DIR")
+        .expect("SODIUM_LIB_DIR must be set because SODIUM_INC_DIR is set. Error");
+    let inc_dir = env::var("SODIUM_INC_DIR")
+        .expect("SODIUM_INC_DIR must be set because SODIUM_LIB_DIR is set. Error");
+
+    print_link();
     println!("cargo:rustc-link-search=native={}", lib_dir);
 
-    inc_dir
+    Some(inc_dir)
 }
 
 /* Must be called when no SODIUM_LIB_DIR and no SODIUM_INC_DIR env vars are set
 This function will set `cargo` flags.
 Return: the first include path from vcpkg
 */
 #[cfg(target_env = "msvc")]
-fn find_libsodium_pkg() -> String {
-    let lib = vcpkg::probe_package("libsodium")
-        .expect("Could not find libsodium on this system");
-    let include_dir = lib.include_paths.get(0)
-        .expect("Could not get includedir using vcpkg");
+fn find_libsodium_pkg() -> Option<String> {
+    let lib = match vcpkg::probe_package("libsodium") {
+        Ok(lib) => lib,
+        Err(e) => {
+            println!("cargo:warning={:?}", e);
+            return None;
+        }
+    };
+    let include_dir = match lib.include_paths.get(0) {
+        Some(dir) => dir,
+        None => {
+            println!("cargo:warning=no include paths from vcpkg");
+            return None;
+        }
+    };
     include_dir.clone().into_os_string().into_string().ok()
-        .expect("Could not get cast include_dir to String")
 }
 
 /* Must be called when no SODIUM_LIB_DIR and no SODIUM_INC_DIR env vars are set
 This function will set `cargo` flags.
 Return: includedir from pkg-config
 */
 #[cfg(not(target_env = "msvc"))]
-fn find_libsodium_pkg() -> String {
+fn find_libsodium_pkg() -> Option<String> {
     let statik = env::var_os("SODIUM_STATIC").is_some();
-    pkg_config::Config::new().statik(statik).probe("libsodium")
-        .expect("Could not find libsodium on this system");
-    pkg_config::get_variable("libsodium", "includedir")
-        .expect("Could not get includedir using pkg-config")
+    if let Err(e) = pkg_config::Config::new().statik(statik).probe("libsodium") {
+        println!("cargo:warning={:?}", e);
+        return None;
+    }
+    pkg_config::get_variable("libsodium", "includedir").ok()
```

Comment:
Still you should warn "Could not get includedir using pkg-config" and return None

Reply:
added